app/javascript/stores/session_store.ts

```diff
@@ -73,10 +130,52 @@ class SessionStore {
   }
 
   @action login(email: string, password: string) {
+    this.passwordState = RequestState.Waiting;
     login(email, password).then(({ data }) => {
       this.setCurrentUser(data);
-    }).catch(() => {
+      this.passwordState = RequestState.Success;
+    }).catch((error) => {
       console.log('Loginfehler!!');
+      this.passwordState = RequestState.Error;
+    });
+  }
+
+  @action setNewPassword(
+    oldPassword: string,
+    newPassword: string,
+    newPasswordConfirmation: string
+  ) {
+    this.passwordState = RequestState.Waiting;
+    setNewPasswordCall(
+      oldPassword,
+      newPassword,
+      newPasswordConfirmation
+    ).then(({ data }) => {
+      this.updateLocalUserCredentials(data);
+      this.user = new User(data);
+      this.passwordState = RequestState.Success;
+    }).catch(() => {
+      this.passwordState = RequestState.Error;
+    });
+  }
+
+  @action reloadUser() {
+    user().then(({ data }) => {
+      const user = this.fetchFromLocalStorage;
+      const updated = _.merge(
+        user,
+        _.pickBy(data, v => v !== null && v !== undefined)
```

Comment:
isn't there loadash helper that matches both cases, not null and not undefined (i thought it was called something like `_.notNil(v)`)

Reply:
couldn't find such a method